### DISCUSSION TITLE
Ability to disable body parsing by default

### DISCUSSION THREAD

**Discussion by agmm** [vercel/next.js#25601] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

Currently it is not possible to disable body parsing by default.

## Describe the solution you'd like

The ability to add an option in `next.config.js` to disable body parsing for all API routes.

One of the possible solutions could be to add a boolean option named `parseBodyByDefault` to specify whether *Next.js* should parse the body of all requests or not.

```js
// next.config.js
module.exports = {
  parseBodyByDefault: false
}
```

## Describe alternatives you've considered

Right now the only alternative that I have found is to add the following code to all files under `pages/api`.

```js
export const config = {
  api: {
    bodyParser: false
  }
};
```

**Comment by brielov** [comment]
Any news on this?

**Comment by blazephoenix** [comment]
Or another option could be to allow disabling it on specific API routes instead of all of them
